From c7b74aa53e9d64da5175f9b86ec7f68e53d1398e Mon Sep 17 00:00:00 2001
From: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Date: Tue, 28 Jul 2020 12:17:30 +0200
Subject: Fix two build failures due to new gcc-10 warnings, such as
 uninitialized variables and duplicated definitions:

e.g. of failure:
/usr/bin/ld: CMakeFiles/gtk-window-decorator.dir/gwd-settings-xproperty-storage.c.o:./obj-x86_64-linux-gnu/gtk/window-decorator/./gtk/window-decorator/gtk-window-decorator.h:269: multiple definition of `window_type_frames'; CMakeFiles/gtk-window-decorator.dir/gtk-window-decorator.c.o:./obj-x86_64-linux-gnu/gtk/window-decorator/./gtk/window-decorator/gtk-window-decorator.h:269: first defined here
/usr/bin/ld: CMakeFiles/gtk-window-decorator.dir/gwd-theme.c.o:./obj-x86_64-linux-gnu/gtk/window-decorator/./gtk/window-decorator/gtk-window-decorator.h:269: multiple definition of `window_type_frames'; CMakeFiles/gtk-window-decorator.dir/gtk-window-decorator.c.o:./obj-x86_64-linux-gnu/gtk/window-decorator/./gtk/window-decorator/gtk-window-decorator.h:269: first defined here
/usr/bin/ld: CMakeFiles/gtk-window-decorator.dir/gwd-theme-cairo.c.o:./obj-x86_64-linux-gnu/gtk/window-decorator/./gtk/window-decorator/gtk-window-decorator.h:269: multiple definition of `window_type_frames'; CMakeFiles/gtk-window-decorator.dir/gtk-window-decorator.c.o:./obj-x86_64-linux-gnu/gtk/window-decorator/./gtk/window-decorator/gtk-window-decorator.h:269: first defined here
/usr/bin/ld: CMakeFiles/gtk-window-decorator.dir/gwd-theme-metacity.c.o:./obj-x86_64-linux-gnu/gtk/window-decorator/./gtk/window-decorator/gtk-window-decorator.h:269: multiple definition of `window_type_frames'; CMakeFiles/gtk-window-decorator.dir/gtk-window-decorator.c.o:./obj-x86_64-linux-gnu/gtk/window-decorator/./gtk/window-decorator/gtk-window-decorator.h:269: first defined here
collect2: error: ld returned 1 exit status
---
 debian/changelog                            | 14 ++++++++++++++
 gtk/window-decorator/gtk-window-decorator.h |  2 +-
 plugins/expo/src/expo.cpp                   |  2 +-
 3 files changed, 16 insertions(+), 2 deletions(-)

#diff --git a/debian/changelog b/debian/changelog
#index 97b93dd..b21831d 100644
#--- a/debian/changelog
#+++ b/debian/changelog
#@@ -1,3 +1,17 @@
#+compiz (1:0.9.14.1+20.10.20200526-0ubuntu3) groovy; urgency=medium
#+
#+  * Initialize 3 arrays to zero to silence a new gcc-10 warning
#+  * Move window_type_frames to extern in the header file, fixing a build
#+    failure with gcc-10
#+
#+ -- Gianfranco Costamagna <locutusofborg@debian.org>  Tue, 28 Jul 2020 11:39:45 +0200
#+
#+compiz (1:0.9.14.1+20.10.20200526-0ubuntu2) groovy; urgency=medium
#+
#+  * Rebuild against new libprotobuf23.
#+
#+ -- Gianfranco Costamagna <locutusofborg@debian.org>  Sun, 26 Jul 2020 10:13:56 +0200
#+
# compiz (1:0.9.14.1+20.10.20200526-0ubuntu1) groovy; urgency=medium
# 
#   [ Daniel Kondor ]
diff --git a/gtk/window-decorator/gtk-window-decorator.h b/gtk/window-decorator/gtk-window-decorator.h
index 6bf5d31..da7f905 100644
--- a/gtk/window-decorator/gtk-window-decorator.h
+++ b/gtk/window-decorator/gtk-window-decorator.h
@@ -266,7 +266,7 @@ typedef struct _default_frame_references
 } default_frame_references_t;
 
 extern default_frame_references_t default_frames[WINDOW_TYPE_FRAMES_NUM * 2];
-const gchar * window_type_frames[WINDOW_TYPE_FRAMES_NUM];
+extern const gchar * window_type_frames[WINDOW_TYPE_FRAMES_NUM];
 
 extern char *program_name;
 
diff --git a/plugins/expo/src/expo.cpp b/plugins/expo/src/expo.cpp
index 86c1618..802071b 100644
--- a/plugins/expo/src/expo.cpp
+++ b/plugins/expo/src/expo.cpp
@@ -708,7 +708,7 @@ ExpoScreen::invertTransformedVertex (const GLScreenPaintAttrib &attrib,
 				     int                       vertex[2])
 {
     GLMatrix sTransform (transform);
-    float    p1[3], p2[3], v[3];
+    float    p1[3] = { 0 } , p2[3] = { 0 }, v[3] = { 0 };
     GLint    viewport[4];
 
     gScreen->glApplyTransform (attrib, output, &sTransform);
-- 
cgit v1.1
